ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity-blocks/activityBlock/activityBlock.component.ts

```diff
@@ -63,8 +65,13 @@ export class ActivityBlockComponent implements OnInit, OnDestroy {
     createChildInstance(): void {
         this.activityServiceAgent.createInstance(this.studyGuid, this.block.activityCode, this.parentActivityInstanceGuid)
             .pipe(
-                concatMap((instanceGuid: ActivityInstanceGuid) => {
-                    return this.activityServiceAgent.getActivitySummary(this.studyGuid, instanceGuid.instanceGuid);
+                tap(response => {
+                    if (response?.blockVisibility) {
+                        this.blockVisibilityChanged.emit(response.blockVisibility);
```

Comment:
should we emit this even if error was catched? if no - I'd suggest to move it to subscription (you can use forJoin operator for example to return multiple observables as a result in concatMap)

Reply:
I believe `tap` operator is not going to be executed if an error occurred and it will go straight to `catchError` operator. Am I wrong?

Reply:
depends on where the error is thrown. If createInstance method throws the error the tap won't be executed, but if getActivitySummary throws the error I believe we execture the tap (before), then go to catchError and then skipping subscription handler

Reply:
Hmm, that's right, if first request succeeds and second one fails the `tap` operator will be executed and it might be confusing for users to see partial updates. I'll rearrange this so that block visibility is emitted if both requests were successful. Thank you!